crates/router/src/connector/dlocal/transformers.rs

```diff
@@ -322,12 +322,12 @@ impl<F, T>
             });
 
         let response = types::PaymentsResponseData::TransactionResponse {
-            resource_id: types::ResponseId::ConnectorTransactionId(item.response.order_id.clone()),
+            resource_id: types::ResponseId::ConnectorTransactionId(item.response.id.clone()),
             redirection_data,
             mandate_reference: None,
             connector_metadata: None,
             network_txn_id: None,
-            connector_response_reference_id: Some(item.response.order_id.clone()),
+            connector_response_reference_id: item.response.order_id.clone(),
```

Comment:
We would like to have fallback incase there is no order id present in respone , we can map this to connector_transaction_id .Please make similar changes in other flows as well 
```suggestion
            connector_response_reference_id: item
                    .response
                    .order_id
                    .or(Some(item.response.id)),
```

Reply:
@srujanchikke 
As the connector_response_reference_id is an optional field, may I kindly inquire about the need to populate the gateway transaction ID as reference_id? Is there a specific reason for this request?

Reply:
Hi @cb-sudheer ,

We are planning to unify the resource ID and connector_response_reference_id in the future. Merchants are encouraged to use our specified ID for searching in both Hyperswitch and Connector. In the event that the Connector lacks a reference ID, we intend to map it using the connector_transaction_id.

Reply:
Yes, you're right this logic can moved to core